tests/Services/SearchServiceTest.php

```diff
@@ -1109,6 +1115,30 @@ public function versionFilterCheckProvider(): array
         ];
     }
 
+    public function testExternalSearchIsNotManipulatedOnNoResultPages()
+    {
+        $requestMock = $this->getMockBuilder(HttpRequest::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $externalSearchServiceMock = $this->getMockBuilder(ExternalSearch::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $originalExternalSearchMock = clone $externalSearchServiceMock;
+
+        $this->requestBuilder->expects($this->once())->method('build')
+            ->willReturn(new Request());
+        $this->client->expects($this->once())
+            ->method('call')
+            ->willReturn($this->getMockResponse('noResults.xml'));
+
+        $searchService = $this->getSearchServiceMock();
+        $searchService->doSearch($requestMock, $externalSearchServiceMock);
+
+        $this->assertEquals($originalExternalSearchMock, $externalSearchServiceMock);
```

Comment:
Can you please add to the test that `$externalSearch->setResults` is not called?